text/0000-process-type-arguments.md

```diff
@@ -0,0 +1,207 @@
+
+This RFC proposes changes to the `launcher`'s handling of args to make the execution of the following process
+```
+[[process]]
+command = "<CMD>"
+args = ["ARG1", "ARG2"]
+```
+produce results equivalent to the execution of `"<CMD>" "<ARG1>" "<ARG2>"` in a shell
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Terminology
+* shell process - a process with `direct=false`
+
+## Philosophy
+The goal is to make the runtime usage of a CNB container as similar as possible to the runtime usage on a non-CNB container, without removing functionality or violating assumptions that have been firmly established in the buildpack community (e.g. defaulting to `web`).
+
+Because the `launcher` must occupy the container `entrypoint`, we will instead use `CNB_PROCESS_TYPE` as a CNB-specific analog to the `entrypoint`. Users can already set `CNB_PROCESS_TYPE` to toggle between processes. Setting `CNB_PROCESS_TYPE=override` will be analogous to clearing the `entrypoint`.
+
+## Behavior
+### Selecting a process type at runtime
+Users will no longer be able to pass a process-type to the launcher as a positional argument. Instead, they must set `CNB_PROCESS_TYPE` in the running container to select a process type.
+If `CNB_PROCESS_TYPE` is unset, it will continue to default to `web`.
+
+The `CNB_PROCESS_TYPE` functionality has existed for a long time and is the only specified way to select a process-type. We will remove the secondary, unspecified way to select a process-type by passing a single argument containing a process-type to the launcher (i.e. `docker run <image> my-process-type` won't work anymore).
+
+### Specifying a custom process at runtime
+If users wish to provide a custom command instead of using one of the buildpack provided process types, they must set `CNB_PROCESS_TYPE=override`. `override` is a special keyword indicating that the entire process shall be defined by the user.
+
+If `CNB_PROCESS_TYPE` is set to `override`, the `launcher` will construct a process from the positional arguments it receives. If the first argument is `--` this will continue to signify a `direct` process. The first (when it is anything other than `--`) or second (if the first is `--`) argument provided to the `launcher`  will become the `command` of the resultant process. Any subsequent arguments to the launcher will become `args` of the resultant process.
+
+Example:
+
+Running `docker run --env CNB_PROCESS_TYPE=override <image> echo hello world`, will be equivalent to running a process type where `direct = false`, `command = "echo"` and `args = ["hello", "world"]`
+
+Running `docker run --env CNB_PROCESS_TYPE=override <image> -- echo hello world`, will be equivalent to running a process type where `direct = true`, `command = "echo"` and `args = ["hello", "world"]`
```

Comment:
What the behavior when I run `docker run -it <image> bash`? An error that no process type is found for "bash"?

Reply:
I am going to resolve this b/c the RFC has changed considerably since this question was asked.